ArduPlane/servos.cpp

```diff
@@ -540,7 +540,21 @@ float Plane::apply_throttle_limits(float throttle_in)
     throttle_watt_limiter(min_throttle, max_throttle);
 #endif
 
-    return constrain_float(throttle_in, min_throttle, max_throttle);
+    auto throttle = constrain_float(throttle_in, min_throttle, max_throttle);
+
+    /*
+      apply TKOFF_THR_MIN if enabled and in an auto-throttle forward
+      transition
+     */
+    if (use_takeoff_throttle_max) {
```

Comment:
Why not apply in the same case as `takeoff_throttle_max`. Presumably because you want to do the TECS reset. But if that is important we should probably be doing the TECS reset for the existing ICE idle, max take off throttle, battery comp and watt limit cases too....

Reply:
also, maybe the bool should be `use_takeoff_throttle_limits` now.